src/mds/CInode.h

```diff
@@ -385,108 +388,104 @@ class CInode : public MDSCacheObject, public InodeStoreBase, public Counter<CIno
   void mark_dirty_rstat();
   void clear_dirty_rstat();
 
-  //bool hack_accessed;
+  //bool hack_accessed = false;
   //utime_t hack_load_stamp;
 
   /**
    * Projection methods, used to store inode changes until they have been journaled,
    * at which point they are popped.
    * Usage:
-   * project_inode as needed. If you're also projecting xattrs, pass
-   * in an xattr map (by pointer), then edit the map.
-   * If you're also projecting the snaprealm, call project_snaprealm after
-   * calling project_inode, and modify the snaprealm as necessary.
+   * project_inode as needed. If you're changing xattrs or sr_t, then pass true
+   * as needed then change the xattrs/snapnode member as needed. (Dirty
+   * exception: project_past_snaprealm_parent allows you to project the
+   * snapnode after doing project_inode (i.e. you don't need to pass
+   * snap=true).
    *
    * Then, journal. Once journaling is done, pop_and_dirty_projected_inode.
    * This function will take care of the inode itself, the xattrs, and the snaprealm.
    */
 
-  struct projected_inode_t {
-    inode_t *inode;
-    std::map<std::string,bufferptr> *xattrs;
-    sr_t *snapnode;
-
-    projected_inode_t()
-      : inode(NULL), xattrs(NULL), snapnode(NULL) {}
-    projected_inode_t(inode_t *in, sr_t *sn)
-      : inode(in), xattrs(NULL), snapnode(sn) {}
-    projected_inode_t(inode_t *in, std::map<std::string, bufferptr> *xp = NULL, sr_t *sn = NULL)
-      : inode(in), xattrs(xp), snapnode(sn) {}
+  class projected_inode {
```

Comment:
please don't rename it to projected_inode. It causes some trouble when rebasing RPs